### DISCUSSION TITLE
enableVersioning=false does not remove existing revisions when resaving entries

### DISCUSSION THREAD

**Discussion by nilsnh** [craftcms/cms#8783] [Ideas] [answered]
### Description

Hi there. We have a FeedMe import that periodically imports entries. After running imports for a while we noticed that versioning was activate and causing unnecessary database bloat.

I deactivated "Enable versioning for entries in this section?" but noticed that there are still many revisions for the imported entry type. I tried to run `./craft resave/entries` but it did not fix it. I guess it's because setting enableVersioning=false skips any revision functionality, even cleanup: https://github.com/craftcms/cms/blob/3990453ebd4693c0c2e74a6274432fe48285db46/src/elements/Entry.php#L1414

I understand that I can run `./craft utils/prune-revisions` and set maxRevisions to a very low number, but that is a global action that would prune revisions for all elements. I would like to only remove revisions for entries where versioning is disabled.

**Proposal for new behavior:**
- Re-saving entries where enableVersioning is deactivated would prune any attached revisions to one or zero if possible.
- Alternatively, maybe the utils/prune-revisions command could be extended with functionality for specifically removing revisions on sections where versioning is disabled.

### Steps to reproduce

1. Create a section type with versioning enabled.
2. Create revisions.
3. Disable versioning and resave entries.
4. Query the database and find that there still exists revisions for the types where it should be deactivated.

Example of a sql query I cobbled together where id is of a resaved entry id '1620773' with versioning disabled.

```sql
select count(*) as total_revisions, r.sourceId, siteId, uri, e.type
from revisions r, elements_sites es, elements e
where r.sourceId = es.elementId and es.elementId = e.id and r.sourceId = '1620773'
group by r.sourceId, siteId, uri, e.type
order by total_revisions DESC;
```

### Additional info

- Craft version: 3.4
- PHP version: 7.2.21
- Database driver & version: mysql 5.7
- Plugins & versions:

**Comment by johnwbaxter** [comment]
Plus one here from me! I've got a matrix table that is up to 2million rows, as well as some other tables and it's not that big a site!!

I just want to remove all those revisions (assuming that's what they are) for that one section, not the whole site and for the revisions to stop building up for that section too.

**Comment by maxstrebel** [comment]
Actually I would even consider this a bug. After deactivating it, your users are still able to revert to an old version.

**Comment by thijskaspers** [comment]
Bump on this. We've got the same situation, had revisions enabled for 2 sections that are updated via Feedme 3 times a day. The elements_sites table is huge now because of this (50 revisions per entry for those 2 sections). We've disabled revisions and hoped that resaving an entry would then prune all old revisions, but it doesn't. The prune revisions console command is useless for us, because that would remove revisions for all sections, which is undesirable.

**Comment by brandonkelly** [comment, marked as answer]
Craft 4.2.0 is out now with a new `--section` option available to the `utils/prune-revisions` command:

```sh
php craft utils/prune-revisions --section=news,about
```

(5d0192d7d58812a68fffa03894f2e1fa7961666f)

**Reply by boudewijn-zicht** [reply]
Is it possible to backport this feature to Craft 3?

**Reply by brandonkelly** [reply]
Just backported it for Craft 3.8 (ad42848145599255da03e00325a0b1adc605d71c).

**Reply by brandonkelly** [reply]
3.8 is out now!
